### PR TITLE
Validate if the openshift master is running with mutitenant network plugin

### PR DESCRIPTION
**Pull request by xiangpengzhao**
If the running network plugin is not OpenShift Multitenant Network Plugin (either redhat/openshift-ovs-subnet or third party plugin), users should not manage pod network.

cc @knobunc Thanks!

### CODE REVIEW

**Comment by ncdc** [comment]
Did you mean to change the file's mode?

**Comment by xiangpengzhao** [comment]
I changed the file's mode by mistake. Fixed. Andy, thanks for pointing it out! @ncdc

File: pkg/cmd/admin/network/project_options.go [hunk 1]
```diff
@@ -84,7 +84,14 @@ func (p *ProjectOptions) Validate() error {
 		errList = append(errList, errors.New("must provide --selector=<project_selector> or projects"))
 	}
 
-	// TODO: Validate if the openshift master is running with mutitenant network plugin
+	clusterNetwork, err := p.Oclient.ClusterNetwork().Get(sdnapi.ClusterNetworkDefault)
+	if err != nil {
+		errList = append(errList, errors.New("could not manage pod network: error on getting the running network plugin"))
```
Comment (pravisankar) [inline pkg/cmd/admin/network/project_options.go]: nit: 'could not manage pod network' is redundant, just throw the error msg something like 'Failed to fetch current network plugin info'

File: pkg/cmd/admin/network/project_options.go [hunk 2]
```diff
@@ -84,7 +84,14 @@ func (p *ProjectOptions) Validate() error {
...
+	} else if !sdnapi.IsOpenShiftMultitenantNetworkPlugin(clusterNetwork.PluginName) {
+		err := fmt.Errorf("could not manage pod network: the running plugin %s is not OpenShift mutitenant network plugin", clusterNetwork.PluginName)
```
Comment (pravisankar) [inline pkg/cmd/admin/network/project_options.go]: 'Using plugin: %q, managing pod network is only supported for openshift multitenant network plugin'?

File: pkg/cmd/admin/network/project_options.go [hunk 3]
```diff
@@ -84,7 +84,14 @@ func (p *ProjectOptions) Validate() error {
 		errList = append(errList, errors.New("must provide --selector=<project_selector> or projects"))
 	}
 
-	// TODO: Validate if the openshift master is running with mutitenant network plugin
```
Comment (pravisankar) [inline pkg/cmd/admin/network/project_options.go]: When we added this CLI functionality, there is no way to fetch network plugin info from the master. Now, ClusterNetwork resource holds the plugin name which enables this validation. 
@xiangpengzhao Thanks for your contribution.

Comment (liggitt) [inline pkg/cmd/admin/network/project_options.go]: Do we expect this to be able to run against an old apiserver as well?

File: pkg/cmd/admin/network/project_options.go (hunk 2 above)
Comment (xiangpengzhao) [inline pkg/cmd/admin/network/project_options.go]: Thanks @pravisankar ! I will fix it as you suggested.

File: pkg/cmd/admin/network/project_options.go (hunk 3 above)
Comment (xiangpengzhao) [inline pkg/cmd/admin/network/project_options.go]: I queried the history of `ClusterNetwork` API and found that it was added on [14 May 2015](https://github.com/openshift/origin/blob/b7d2fe8a360c0730c49f48de467180a78318bdad/pkg/client/client.go#L132). It means that an apiserver before that day would not support `ClusterNetwork` API. So the question is, do we expect the functionality to be able to run against such an old apiserver? :)

Comment (pravisankar) [inline pkg/cmd/admin/network/project_options.go]: If the user is not using openshift-sdn plugin then "default" cluster network resource won't be present and this validation will be incorrect. So if the error returned from Get(sdnapi.ClusterNetworkDefault) is kerrs.IsNotFound or if the plugin is not openshift multitenant, we should throw an error with message 'managing pod network is only supported for openshift multitenant network plugin'.

For the second issue, AFAIK we don't expect old api server to return ClusterNetwork resource without pluginName as we only allow incremental upgrades i.e. 3.2 to 3.3 and then 3.3 to 3.4 release (https://github.com/openshift/origin/pull/11073#issuecomment-250283774). PluginName field was added in 3.3 release, so this validation in 3.4 can expect valid plugin name if present.

**Comment by xiangpengzhao** [comment]
Squashed. Thanks @pravisankar

**Comment by knobunc** [comment]
[merge]

**Comment by knobunc** [comment]
Flake https://github.com/openshift/origin/issues/11004 [merge]

**Pull request merged**